events/dbpersist.go

```diff
@@ -137,24 +314,51 @@ func (p *DbPersistence) Playback(ctx context.Context, since int64, cb func(*XRPC
 }
 
 func (p *DbPersistence) uidForDid(ctx context.Context, did string) (util.Uid, error) {
+	if uid, ok := p.didCache.Get(did); ok {
+		return uid.(util.Uid), nil
+	}
+
 	var u models.ActorInfo
 	if err := p.db.First(&u, "did = ?", did).Error; err != nil {
 		return 0, err
 	}
 
+	p.didCache.Add(did, u.Uid)
+
 	return u.Uid, nil
 }
 
 func (p *DbPersistence) didForUid(ctx context.Context, uid util.Uid) (string, error) {
+	if did, ok := p.uidCache.Get(uid); ok {
+		return did.(string), nil
+	}
+
 	var u models.ActorInfo
 	if err := p.db.First(&u, "uid = ?", uid).Error; err != nil {
 		return "", err
 	}
 
+	p.uidCache.Add(uid, u.Did)
+
 	return u.Did, nil
 }
 
-func (p *DbPersistence) hydrateRepoEvent(ctx context.Context, rer *RepoEventRecord) (*comatproto.SyncSubscribeRepos_Commit, error) {
+func (p *DbPersistence) hydrateHandleChange(ctx context.Context, rer *RepoEventRecord) (*XRPCStreamEvent, error) {
+	did, err := p.didForUid(ctx, rer.Repo)
+	if err != nil {
+		return nil, err
+	}
+
+	return &XRPCStreamEvent{
+		RepoHandle: &comatproto.SyncSubscribeRepos_Handle{
+			Did:    did,
+			Handle: *rer.NewHandle,
+			Time:   rer.Time.Format(util.ISO8601),
+		},
+	}, nil
+}
+
+func (p *DbPersistence) hydrateCommit(ctx context.Context, rer *RepoEventRecord) (*XRPCStreamEvent, error) {
```

Comment:
lets sanity check that the rer.Commit isnt nil at some point in this function, now that we've changed it to being a pointer